app/services/messages/service.go

```diff
@@ -385,17 +390,27 @@ func (ss *Service) VerifyEthereumTxAuthenticity(tm encoding.TopicMessage) (bool,
 		return false, nil
 	}
 
+	valid, _, err := ss.contractsService.IsValidBridgeAsset(nil, dbTx.SourceAsset)
+	if err != nil {
+		ss.logger.Errorf("Could not validate provided asset [%s] - Error: [%s]", dbTx.SourceAsset, err)
+		return false, err
+	}
+	if !valid {
+		ss.logger.Errorf("Provided Asset is not supported - [%s]", dbTx.SourceAsset)
+		return false, err
+	}
```

Comment:
There is no point in validating the asset from the database. You should validate that the `erc20address` from the `DecodeBridgeMintFunction` is equal to the `TargetAsset` in the database